src/bp/core/services/action/action-service.ts

```diff
@@ -37,9 +47,27 @@ export type ActionDefinition = {
 }
 
 export class ScopedActionService {
-  constructor(private ghost: GhostService, private logger, private botId: string) {}
+  private _actionsCache: ActionDefinition[] | undefined
+  private _scriptsCache: Map<string, string> = new Map()
+
+  constructor(private ghost: GhostService, private logger, private botId: string, private cache: ObjectCache) {
+    this._listenForCacheInvalidation()
+  }
+
+  private _listenForCacheInvalidation() {
+    this.cache.events.on('invalidation', key => {
+      if (key.toLowerCase().indexOf(`${path.sep}actions`) > -1) {
```

Comment:
nice (`path.sep`) ... makes me think that's an issue in my other cache invalidations

Reply:
Maybe, I only checked actions and hooks